sys/include/xtimer.h

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2015 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup  sys_xtimer Timers
+ * @ingroup   sys
+ * @brief     Provides a high level timer module to register
+ *            timers, get current system time, and let a thread sleep for
+ *            a certain amount of time.
+ *
+ * The implementation takes one low-level timer that is supposed to run at 1MHz
+ * speed and multiplexes it.
+ *
+ * Insertion and removal of timers has O(n) complexity with (n) being the
+ * number of active timers.  The reason for this is that multiplexing is
```

Comment:
> Most likely the *TSCH* implementation (currently this is taken from *OpenWSN*) will use `periph/timers`.

Your *post* reads *like* one of these modern *comics*. ;)

Probably TSCH will run fine even if using xtimers.
And, xtimer will not become deterministic just because the number of
timers is limited.

xtimer does deactivate interrupts while adding / removing timers, but
even with a hundred running timers in one of the lists, the time
interrupts are disabled is probably <1us.

Could you elaborate on the timing parameters of TSCH?